pkg/apis/core/types_shoot.go

```diff
@@ -428,6 +432,30 @@ type AdmissionPlugin struct {
 	Config *runtime.RawExtension
 }
 
+// WatchCacheSizes contains configuration of the API server's watch cache sizes.
+type WatchCacheSizes struct {
+	// Default configures the default watch cache size of the kube-apiserver
+	// (flag `--default-watch-cache-size`, defaults to 100).
+	// See: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/
+	Default *int32
+	// Resources configures the watch cache size of the kube-apiserver per resource
+	// (flag `--watch-cache-sizes`).
+	// See: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/
+	Resources []ResourceWatchCacheSize
+}
+
+// ResourceWatchCacheSize contains configuration of the API server's watch cache size for one specific resource.
+type ResourceWatchCacheSize struct {
+	// APIGroup is the API group of the resource for which the watch cache size should be configured.
+	// An unset value is used to specify the legacy core API (e.g. for `secrets`).
+	APIGroup string
```

Comment:
```suggestion
	APIGroup *string
```

It is optional, right?

Reply:
Yes, it's optional. And it even works that way, as empty strings are omitted on `omitempty`.
Do I still have to use a pointer for optional strings according to the API conventions?

Reply:
I would vote for 'yes', if there is no strong argument against it. :)

Reply:
Ah, found it https://github.com/gardener/gardener/pull/2668
Will adapt :)

Reply:
Done